### DISCUSSION TITLE
Separate out isEnabled and isEditable flags

### DISCUSSION THREAD

**Discussion by iliasaz** [krzyzanowskim/STTextView#24] [Questions]
Hi Marcin,

In STTextView, you have this code:

```
/// A Boolean value that controls whether the text view allows the user to edit text.
    @Invalidating(.insertionPoint, .cursorRects)
    @objc dynamic open var isEditable: Bool = true {
        didSet {
            isSelectable = isEditable
        }
    }
```

I have a use case where the view should be read-only for user inputs, but I'd like to be able to select/copy the text from the view and to format/highlight the text within the view (ex., TreeSitter.) It would be great if we could control these behaviors independently of each other. 

Thanks!

**Comment by krzyzanowskim** [comment]
These settings has to be connected, however not necessary in that way. I changed it now on `main` branch

either way, I belive you can archive selectable, non-editable by setting it like this:

```swift
textView.isEditable = false
textView.isSelectable = true
```

**Reply by zeroskylian** [reply]
Hello. When I set `isEditable  = false`,  `isSelectable  = true`, I can't select any text.  Is there any other operation?
iOS 17 Device

**Reply by cxa** [reply]
https://github.com/krzyzanowskim/STTextView/blob/267c783764fd70b0ebd19eff0e111c0079b605b7/Sources/STTextViewUIKit/STTextView.swift#L438-L440

Solution: remove `&& isEditable`.

**Reply by zeroskylian** [reply]
> https://github.com/krzyzanowskim/STTextView/blob/267c783764fd70b0ebd19eff0e111c0079b605b7/Sources/STTextViewUIKit/STTextView.swift#L438-L440
> 
> Solution: remove `&& isEditable`.

Thanks!

**Reply by krzyzanowskim** [reply]
wai, is that different between uikit and appkit version?

**Reply by zeroskylian** [reply]
> wai, is that different between uikit and appkit version?

I think so, mac Appkit work well with `isEditable  = false, isSelectable  = true`, but it does not work in ios and Mac catalyst

**Reply by krzyzanowskim** [reply]
thaks. in that case the UIKit should follow the AppKit behavior. Do you want to make a PR with the fix for that by any chance?

**Reply by zeroskylian** [reply]
@cxa

**Reply by cxa** [reply]
PR sent.
